cognitive/src/test/scala/com/microsoft/azure/synapse/ml/cognitive/split1/OpenAISuite.scala

```diff
@@ -23,34 +20,77 @@ class OpenAICompletionSuite extends TransformerFuzzing[OpenAICompletion] with Op
 
-  lazy val completion: OpenAICompletion = new OpenAICompletion()
-    .setSubscriptionKey(openAIAPIKey)
-    .setDeploymentName("text-davinci-001")
-    .setServiceName(openAIServiceName)
-    .setMaxTokens(20)
-    .setLogProbs(5)
-    .setPromptCol("prompt")
-    .setOutputCol("out")
+  lazy val promptCompletion: OpenAICompletion = newCompletion.setPromptCol("prompt")
+  lazy val bulkPromptCompletion: OpenAICompletion = newCompletion.setBulkPromptCol("bulkPrompt")
+  lazy val indexCompletion: OpenAICompletion = newCompletion.setIndexPromptCol("index")
+  lazy val bulkIndexCompletion: OpenAICompletion = newCompletion.setBulkIndexPromptCol("bulkIndex")
 
-  lazy val df: DataFrame = Seq(
+  lazy val promptDF: DataFrame = Seq(
     "Once upon a time",
     "Best programming language award goes to",
     "SynapseML is "
   ).toDF("prompt")
 
+  lazy val bulkPromptDF: DataFrame = Seq(
+    Seq(
+      "Now is the time",
+      "Knock, knock",
+      "Ask not")
+  ).toDF("bulkPrompt")
+
+  lazy val indexDF: DataFrame = Seq(
+    Seq(1212, 318, 247, 1332)
+  ).toDF("index")
+
+  lazy val bulkIndexDF: DataFrame = Seq(
+    Seq(
+      Seq(1212, 318, 257, 1332),
+      Seq(1334, 259, 320, 1214))
+  ).toDF("bulkIndex")
+
   test("Basic Usage") {
+    testCompletion(promptCompletion, promptDF, 10)
+  }
+
+  test("Bulk Prompt") {
+    testCompletion(bulkPromptCompletion, bulkPromptDF, 20)
+  }
+
+  test("Index") {
+    testCompletion(indexCompletion, indexDF, 10)
+  }
+
+  test("Bulk Index") {
+    testCompletion(bulkIndexCompletion, bulkIndexDF, 20)
+  }
+
+  def testCompletion(completion: OpenAICompletion, df: DataFrame, requiredLength: Int): Unit = {
     val fromRow = CompletionResponse.makeFromRowConverter
-//    completion.transform(df).collect().map(r =>
-//      fromRow(r.getAs[Row]("out")).choices.head.text.length > 10)
-    completion.transform(df).show(truncate=false)
+    val transformed = completion.transform(df)
+    //transformed.show(truncate=false) // uncomment for debugging
+    transformed.collect().map(r =>
+      assert(fromRow(r.getAs[Row]("out")).choices.head.text.length > requiredLength))
+  }
+
+  def newCompletion(): OpenAICompletion = {
+    new OpenAICompletion()
+      .setSubscriptionKey(openAIAPIKey)
+      .setDeploymentName("text-davinci-001")
+      .setServiceName(openAIServiceName)
+      .setMaxTokens(20)
+      .setLogProbs(5)
+      .setOutputCol("out")
   }
 
   override def assertDFEq(df1: DataFrame, df2: DataFrame)(implicit eq: Equality[DataFrame]): Unit = {
     super.assertDFEq(df1.drop("out"), df2.drop("out"))(eq)
   }
 
-  override def testObjects(): Seq[TestObject[OpenAICompletion]] =
-    Seq(new TestObject(completion, df))
+  override def testObjects(): Seq[TestObject[OpenAICompletion]] = Seq(
+    new TestObject(promptCompletion, promptDF),
+    new TestObject(bulkPromptCompletion, bulkPromptDF),
+    new TestObject(indexCompletion, indexDF),
+    new TestObject(bulkIndexCompletion, bulkIndexDF))
```

Comment:
Dont necessarily need all these in test objects if you don't want! This will cause all instances to generate python tests and serialization tests